client/state/posts/selectors.js

```diff
@@ -20,10 +32,40 @@ import { DEFAULT_POST_QUERY } from './constants';
  * @return {Object}          Post object
  */
 export function getPost( state, globalId ) {
-	return state.posts.items[ globalId ];
+	return getPostsByGlobalId( state )[ globalId ];
+}
+
+/**
+ * Returns an array of all posts.
+ *
+ * @param  {Object} state    Global state tree
+ * @return {Array}           Posts array
+ */
+export function getPosts( state ) {
+	return state.posts.items;
 }
 
 /**
+ * Returns a mapping of site ID, post ID pairing to global post ID.
+ *
+ * @param state
+ */
+export const getSitePosts = createSelector(
```

Comment:
I think it makes most sense to keep it in the selectors file. If something seems like magic, we should take the time to read the source :smile:  https://github.com/reactjs/reselect/blob/master/src/index.js